library/src/main/java/com/bumptech/glide/load/resource/bitmap/TransformationUtils.java

```diff
@@ -98,6 +98,49 @@ public static Bitmap centerCrop(@NonNull BitmapPool pool, @NonNull Bitmap inBitm
   }
 
   /**
+   * A potentially expensive operation to crop the given Bitmap so that it fills the given
+   * dimensions. This operation is significantly less expensive in terms of memory if a mutable
+   * Bitmap with the given dimensions is passed in as well.
+   *
+   * @param pool     The BitmapPool to obtain a bitmap from.
+   * @param inBitmap   The Bitmap to resize.
+   * @param width    The width in pixels of the final Bitmap.
+   * @param height   The height in pixels of the final Bitmap.
+   * @param xPercentage The horizontal percentage of the crop. 0.0f => left, 0.5f => center, 1.0f => right or anything in between 0 and 1
+   * @param yPercentage The vertical percentage of the crop. 0.0f => top, 0.5f => center, 1.0f => bottom or anything in between 0 and 1
+   * @return The resized Bitmap (will be recycled if recycled is not null).
+   */
+  public static Bitmap cropPosition(@NonNull BitmapPool pool, @NonNull Bitmap inBitmap, int width,
+      int height, float xPercentage, float yPercentage) {
+    if (inBitmap.getWidth() == width && inBitmap.getHeight() == height) {
+      return inBitmap;
+    }
+    // From ImageView/Bitmap.createScaledBitmap.
+    final float scale;
+    float dx = 0, dy = 0;
+    Matrix m = new Matrix();
```

Comment:
There are an awful lot of `0.5`s hardcoded here, it feel suspicious with 0.5 meaning center, but also having x/yPercentage.